Add an Outline

code3z opened the pull request:
Here we can discuss what exactly we want to write about, and organize an outline 👇

Commit:
Create outline.md

code3z commented:
Also: please read the outline before commenting

tiosgz commented on outline.md:
```diff
@@ -0,0 +1,39 @@
+- Inspecting the interface (3b)
+ - And using the Console
+- Your First JS Mod (5)
+- Keyboard Shortcuts (9)
```
What for? *Modding* keyboard shortcuts, or some (?) ‘standard set’ that might be useful?

code3z replied:
Lonm’s keyboard machine is very useful for modding because you can call your own functions. It wouldn’t take long to explain.

tiosgz replied:
I’d move it into *advanced JS* then.

code3z replied:
👌

tiosgz commented on outline.md:
```diff
@@ -0,0 +1,39 @@
+## Advanced JS
+- Mutation Observers (11)
+- Vivaldi. APIs (12)
+- Mod UI and adding
```
?

tiosgz reviewed (commented):
I am not sure if it makes sense at all, but you seem to see a good reason for it. Except for the usual handling (close or merge), this PR might as well be kept open – what do you think? (It has at least the downside that it is you who has to do any changes to the outline.)

Because of the above, I’ll leave writing&code style be for now.

code3z committed:
Update outline.md

code3z committed:
Mutation Observer link

code3z committed:
Outline typo

code3z committed:
Update outline.md

code3z commented:
> this PR might as well be kept open – what do you think?

I think, until the outline is done we should leave it open. I will also create some empty template files once we have a good outline.

More changes to the outline can always be a seperate pr

Zalext commented:
That Index is why I've talking about.

A Road Map first, then elaborate each point.


👍

code3z commented:
@Zalext Do you have any opinions on what should or shouldn't be included?

Zalext commented:
> @Zalext Do you have any opinions on what should or shouldn't be included?

It's Ok,
Seems that contains all the needed, but since I do not work as Dev, mine is an objective view and may miss something.

tiosgz commented:
> A Road Map first, then elaborate each point.

I see it the other way around – write what we have in mind; when we find out that something is missing, either write it right away or note it somewhere. As can be seen in #4, even a ‘simple’ thing can get complex.

Zalext commented:
Yes,

I don't see the point in do it reversed, but Ok.

code3z commented:
> write what we have in mind; when we find out that something is missing, either write it right away or note it somewhere

Yes. Down with organization!

But - didn't you create this repo because you thought we were jumping into things too quickly?

tiosgz commented:
> But - didn't you create this repo because you thought we were jumping into things too quickly

As I see/saw it, you were about to just write something & accept whatever contributions unless they were completely wrong. I want to accept guides that are complete and read by more people before publishing, see how much you reminded me of in #4. Because of this it is not so easy to guess the approximate size of each of the guides, so I think it is better to keep things flexible and decide when they come.

And again, getting to this PR: does it make sense? The list of contents should reflect the current state, as I started doing in the `README` in #4. The list of missing guides should not include the written ones, as GitHub issues can do.

code3z commented:
@tiosgz I was going to convert the outline into files. Should I convert them into issues instead?

tiosgz commented:
Hmm, yes, you can. I’d say that 5 & 6 belong into one issue and 7 & 10 are already in #5.

code3z commented:
> 5 & 6 belong into one issue

No, I think they are separate, we have great CSS modders who can’t write any JS.



> and 7 & 10 are already in #5.

Should I combine 7 & 10? I thought of them as a bit separate but I guess they could be the same.

tiosgz commented:
> > 5 & 6 belong into one issue
> 
> No, I think they are separate, we have great CSS modders who can’t write any JS.

Wrt writing, they are separate. Wrt how I think about them, they are just ‘example mods’. But do it as you will, I don’t care so much.

Zalext commented:
Hi,

On External Resources, this comment, made on the wrong place, be useful.

> Forgot to mention on to add some useful Tools.
> 
> On my side, for example on Win, Notepad++ as a simple but powerful tool.
> Even has a PortableApps.com version.
> 
> For Linux I don't know, seen VIM and others most related to DE but I don't know what would be suggested.
> 
> Dev tools as Atom, Sublime and others may be mentioned as a next step for those interested in create new ones and starts from here [V Modding Guide] without previous Dev surface.

code3z commented:
@Zalext You can comment here:
https://github.com/tiosgz/modding-vivaldi/issues/5

code3z commented:
Is it okay if I post about this on the forum asking if anyone would like to help? (Is that a bad idea?) Is there anyone on the forum you want to ask if they would like to help?

Zalext commented:
> Is it okay if I post about this on the forum asking if anyone would like to help? (Is that a bad idea?) Is there anyone on the forum you want to ask if they would like to help?

Maybe using the original Topic asking about all that and pointing here or opening a new one pointing to original on forum to understand where it comes from and here.

tiosgz commented:
> @Zalext You can comment here:
> #5

Actually, what they suggest is for another issue. (Is this what you meant by that generic ‘external resources’ in the outline? I understood it more like resources for learning.)

> Is it okay if I post about this on the forum asking if anyone would like to help? (Is that a bad idea?) Is there anyone on the forum you want to ask if they would like to help

I think most people who could potentionally want to help already know about this. There’re only Ornorm & hlehyaric, who said something like they didn’t want an account on GitHub (but I’ll ask them in PM just in case).

code3z committed:
Merge branch 'tiosgz:master' into Outline

Zalext commented:
Well,

Just seen External Resources.

It would be just for Learning and add those suggestions somewhere else or append it as Suggested Tools as well.

Whichever better.

code3z commented:
Either way. Or we can combine the two.

Commit:
Add file structure

Added blank files for the high-priority components of the guide. They should also have accompanying github issues.

code3z commented:
I think this should be merged, as it provides important file structure to organize the repo and should help get us started.

ortiza5 reviewed (changes requested):
`creating-mods/js/resources,md` should probably be `creating-mods/js/resources.md` (".md" instead of ",md") before merging

code3z commented:
@ortiza5 Fixed, thanks!

tiosgz reviewed (changes requested):
I wasn’t a big fan of this PR even in the beginning (but that was *entirely* my personal opinion about organising everything), but I am strongly against it now that you have added all the files. It is just adding unnecessary work as soon as something turns out to be better differently than how you’ve drafted it. Can you give me a reason why it is good?

code3z commented:
@tiosgz Sorry, I thought of the blank files as part of the outline. I can take them out. Or should we manage the project entirely from GitHub issues?

Would you be okay with adding a basic folder structure?

tiosgz commented:
> Sorry, I thought of the blank files as part of the outline. I can take them out.

Yes, that would be better IMO.

> Or should we manage the project entirely from GitHub issues?

Depends – it seems like the better option to me, but you seem to like this way & IIRC (cannot find it now) @ortiza5 also said something like that, so we can have both.

> Would you be okay with adding a basic folder structure

I would rather avoid that as well. Anyway, I think that git ignores folders if they don’t contain files.

ortiza5 commented:
> Or should we manage the project entirely from GitHub issues?

Do we want to add the Kanban style project management feature to this repo? That would give a quick way to view what people are working on and what stage of progress they are in.

code3z commented:
Yes! I forgot about that.
How about this:
I'll convert the outline to kanban.
@tiogz makes the basic folder structure (creating-mods, using-mods, js, css directories, whatever works well) leaving enough room for flexibility (we won't make all the template files)
I'll close this pull request and start working on sharing mods and the welcome and docsify and devtools or something useful

tiosgz commented:
> Do we want to add the Kanban style project management feature to this repo? That would give a quick way to view what people are working on and what stage of progress they are in

Do you mean something more than GH issues can do? They can have an assignee; PRs can be drafts / ready for review; when something’s done, it’s closed&merged. I may be blind, but what is missing / not good enough?

> `@tiogz` makes the basic folder structure (creating-mods, using-mods, js, css directories, whatever works well) leaving enough room for flexibility (we won't make all the template files)

Again, I think that it should be done when it’s needed, not sooner.

> I'll close this pull request and start working on sharing mods and the welcome and docsify and devtools or something useful

I recommend you to choose only one & continue when it’s done 😉

ortiza5 commented:
> Do you mean something more than GH issues can do? They can have an assignee; PRs can be drafts / ready for review; when something’s done, it’s closed&merged. I may be blind, but what is missing / not good enough?

It is just a visualization tool for issues. Ideally it should be set up with automations to not need any manually moving of issues or pull requests around. Watch [this](https://www.youtube.com/watch?v=ff5cBkPg-bQ) video to see how it can be set up (the automations that were skipped should probably be added).

tiosgz commented:
Yay, YT. Okay, will check that tomorrow.

Commit:
typo

tiosgz commented:
Project set up (or I suppose so). Is there anything left in this PR?

code3z commented:
There are some issues based on the outline that need to be opened. After that I will close it.

code3z commented:
Thanks for setting up projects!

code3z commented:
Are there any other columns that would help in projects? Maybe "requires discussion," "low priority," or "long term"?

tiosgz commented:
> Are there any other columns that would help in projects?

That would require manual messing with it, wouldn’t it? 😁 (I’m not against it)

> Maybe "requires discussion," "low priority," or "long term"

Could be… but I prefer labels & the rule set doesn’t allow me to automate moving to columns based on labels.

code3z commented:
Everything should be good now.
